### DISCUSSION TITLE
73. 조인으로 인해 성능 저하가 발생할 경우, 이를 해결하기 위한 방법은 무엇인가요?

### DISCUSSION THREAD

**Discussion by cutehumanS2** [jewelry-box/cs-interview#75] [General]
조인으로 인해 성능 저하가 발생할 경우, 이를 해결하기 위한 방법은 무엇인가요?

**Comment by eun-byeol** [comment]
`인덱스`, `필터링 조건`, `서브쿼리` 사용을 고려할 수 있습니다. 해결되지 않는 경우 `역정규화`를 통해 조인을 줄일 수 있습니다.
조인에 사용되는 칼럼에 적절한 인덱스를 생성하여, 인덱스로 데이터 접근 시간을 줄일 수 있습니다.
WHERE 절이나 서브쿼리를 사용해 조인 전에 데이터를 추출하여, 불필요한 데이터 조인을 제거할 수 있습니다.
정규화된 DB를 역정규화하여 조인 수를 줄여 읽기 성능을 높일 수 있습니다.
